pkg/commands/componentarchive/resources/add.go

```diff
@@ -314,7 +314,8 @@ func generateResourcesFromReader(cd *cdv2.ComponentDescriptor, reader io.Reader)
 }
 
 func (o *Options) addInputBlob(fs vfs.FileSystem, archive *ctf.ComponentArchive, resource *InternalResourceOptions) error {
-	blob, err := resource.Input.Read(fs, resource.Path)
+	includeRootDir := resource.Type == "helm"
```

Comment:
I would not make it specific to helm (also the helm resource type was updated to `helm.io/chart` to conform with the cnudie recommendation).
I would rather extend the input blob type so that this options can be configurable per resource.
e.g. so that a possible source definition could look like (not sure about the `preserveDir`, this is just an idea. Maybe it could be even more generic by passing dedicated tar options )
```
name: my-helmm-chart
...
input:
  type: "dir"
  path: "./my-chart"
  preserveDir: true
```

Reply:
We have implemented your proposal and added a boolean `preserveDir` to the input structure.